### PR TITLE
Update F# web templates to be more idiomatic

### PR DESCRIPTION
**Pull request by cartermp**
I noticed a few things about the current templates that could probably improve.

Separately, would it be okay if I added a Worker template? I manually created one here and it seems to work:

https://github.com/cartermp/fsharpworker

(the ID in the project properties was just copypasta from a C# one, I assume this would not be identical)

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-FSharp/WeatherForecast.fs [hunk 1]
```diff
@@ -3,9 +3,9 @@ namespace Company.WebApplication1
 open System
 
 type WeatherForecast =
-    { Date: DateTime
-      TemperatureC: int
-      Summary: string }
+  { Date: DateTime
```
Comment (cartermp) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-FSharp/WeatherForecast.fs:6]: oops the tabbing here got funky, will fix

**Comment by mkArtakMSFT** [comment]
@pranavkm can you please review this? Thanks!

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-FSharp/Startup.fs [hunk 2]
```diff
@@ -14,32 +14,29 @@ open Microsoft.Extensions.Configuration
...
 
     // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-    member this.Configure(app: IApplicationBuilder, env: IWebHostEnvironment) =
+    member _.Configure(app: IApplicationBuilder, env: IWebHostEnvironment) =
         if (env.IsDevelopment()) then
             app.UseDeveloperExceptionPage() |> ignore
 #if (!NoHttps)
-
-        app.UseHttpsRedirection() |> ignore
+        app.UseHttpsRedirection()
```
Comment (pranavkm) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-FSharp/Startup.fs:30]: @javiercn what do you think about the fluent-ness? This doesn't use auth, so there are only two code paths to manage here.

**Review by pranavkm — APPROVED** [review]

**Comment by pranavkm** [comment]
> Separately, would it be okay if I added a Worker template? I manually created one here and it seems to work:

/cc @shirhatti

**Comment by shirhatti** [comment]
Yes please! @bradygaster

**Comment by cartermp** [comment]
Cool, I'll yeet one over in a PR today

**Comment by pranavkm** [comment]
Thanks @cartermp

**Pull request merged**